tests/openapi/client/openapi_task_check

```diff
@@ -0,0 +1,211 @@
+            ],
+        },
+    }
+    resp = requests.post(url=API_ENDPOINT, json={"remove_meta": True, "task": task})
+    print("start_task_failed resp=", resp.json())
+    assert resp.status_code == 400
+
+
+def start_noshard_task_success():
+    task = {
+        "name": NO_SHARD_TASK_NAME,
+        "task_mode": "all",
+        "shard_mode": "pessimistic",
+        "meta_schema": "dm-meta",
+        "enhance_online_schema_change": True,
+        "on_duplicate": "error",
+        "target_config": {
+            "host": "127.0.0.1",
+            "port": 4000,
+            "user": "root",
+            "password": "",
+        },
+        "table_migrate_rule": [
+            {
+                "source": {
+                    "source_name": SOURCE1_NAME,
+                    "schema": "openapi",
+                    "table": "*",
+                },
+                "target": {"schema": "openapi", "table": ""},
+            },
+            {
+                "source": {
+                    "source_name": SOURCE2_NAME,
+                    "schema": "openapi",
+                    "table": "*",
+                },
+                "target": {"schema": "openapi", "table": ""},
+            },
+        ],
+        "source_config": {
+            "source_conf": [
+                {"source_name": SOURCE1_NAME},
+                {"source_name": SOURCE2_NAME},
+            ],
+        },
+    }
+    resp = requests.post(url=API_ENDPOINT, json={"remove_meta": True, "task": task})
+    print("start_noshard_task_success resp=", resp.json())
+    assert resp.status_code == 201
+
+
+def start_shard_task_success():
+    task = {
+        "name": SHARD_TASK_NAME,
+        "task_mode": "all",
+        "shard_mode": "pessimistic",
+        "meta_schema": "dm-meta",
+        "enhance_online_schema_change": True,
+        "on_duplicate": "error",
+        "target_config": {
+            "host": "127.0.0.1",
+            "port": 4000,
+            "user": "root",
+            "password": "",
+        },
+        "table_migrate_rule": [
+            {
+                "source": {
+                    "source_name": SOURCE1_NAME,
+                    "schema": "openapi",
+                    "table": "*",
+                },
+                "target": {"schema": "openapi", "table": "t"},
+                "event_filter_name": ["rule-1"],
+            },
+            {
+                "source": {
+                    "source_name": SOURCE2_NAME,
+                    "schema": "openapi",
+                    "table": "*",
+                },
+                "target": {"schema": "openapi", "table": "t"},
+                "event_filter_name": ["rule-2"],
+            },
+        ],
+        "source_config": {
+            "source_conf": [
+                {"source_name": SOURCE1_NAME},
+                {"source_name": SOURCE2_NAME},
+            ],
+        },
+        "binlog_filter_rule": [
+            {
+                "rule_name": "rule-1",
```

Comment:
not sure if the rule_name should be the key of this element, which means the binlog_filter_rule should be a map not list.

Reply:
yes, and we cau use openapi3.0 [additionalProperties](https://swagger.io/docs/specification/data-models/dictionaries/) to impl this, added in https://github.com/pingcap/dm/pull/2141/commits/8e9565919ac883c762ea17edd0831dc56bcc3953